frontend/src/metabase/dashboard/components/Dashboard/Dashboard.styled.tsx

```diff
@@ -164,8 +165,8 @@ export const ParametersAndCardsContainer = styled.div<{
 
   padding-bottom: 40px;
   /* Makes sure it doesn't use all the height, so the actual content height could be used in embedding #37437 */
-  align-self: ${({ shouldMakeDashboardHeaderStickyAfterScrolling }) =>
-    !shouldMakeDashboardHeaderStickyAfterScrolling && "flex-start"};
+  align-self: ${({ shouldMakeDashboardHeaderStickyAfterScrolling, isEmpty }) =>
+    !shouldMakeDashboardHeaderStickyAfterScrolling && !isEmpty && "flex-start"};
```

Comment:
This was the trickiest part of getting the layout "right".
The highlighted code was put in place for a specific reason of fixing an issue and I knew I cannot change it. But shrinking the parent (this element) meant that I can't extend any of its children to the 100% of the viewport height.

In the end I realized that it should be perfectly safe to exclude the styling when the container is empty. I'd still appreciate someone from @metabase/embedding to confirm this is safe.

Btw, this is an example of a VERY useful comment in the source code. Can't say enough how much I appreciated the context it provides!